src/main/scala/PropertyChecker.scala

```diff
@@ -402,4 +412,84 @@ object PropertyChecker {
        (objectNode, warnings, csvwPropertyType)
     }
   }
+
+  def tableSchemaProperty(typeString: PropertyType.Value): (JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    def tableSchemaPropertyInternal(value: JsonNode, baseUrl: String, lang: String): (JsonNode, Array[String], PropertyType.Value) = {
+      var schemaBaseUrl = new URL(baseUrl)
+      var schemaLang = lang
+
+      var schemaJson: ObjectNode = null
+      if(value.isTextual) {
+        val schemaUrl = new URL(new URL(baseUrl), value.asText())
+        val objectMapper = new ObjectMapper()
+        schemaJson = objectMapper.readTree(schemaUrl).asInstanceOf[ObjectNode]
+        if (!schemaJson.path("@id").isMissingNode) {
+          // Do something here as object node put method doesn't allow uri object
+          val absoluteSchemaUrl = new URL(schemaUrl, schemaJson.get("@id").asText())
+          schemaJson.put("@id", absoluteSchemaUrl.toString)
+        } else {
+          schemaJson.put("@id", schemaUrl.toString)
+        }
+        val (newSchemaBaseUrl, newSchemaLang) = fetchSchemaBaseUrlAndLangAndRemoveContext(schemaJson, schemaBaseUrl, schemaLang)
+        schemaBaseUrl = newSchemaBaseUrl
+        schemaLang = newSchemaLang
+      } else if(value.isObject) {
+        schemaJson = value.deepCopy()
+      } else {
+        return (NullNode.getInstance(), Array[String]("invalid_value"), PropertyType.Table)
+      }
+
+      var warnings = Array[String]()
+      val fieldsAndValues = Array.from(schemaJson.fields.asScala)
+      for(fieldAndValue <- fieldsAndValues) {
+        val p = fieldAndValue.getKey
```

Comment:
The function is getting a bit long. I'd be tempted to pull the code inside this for loop out into a separate function.

Reply:
Done